documentation/src/main/asciidoc/topics/con_performance_metric_considerations.adoc

```diff
@@ -0,0 +1,23 @@
+:_content-type: CONCEPT
+[id='performance-metric-considerations_{context}']
+= Performance metric considerations
+
+{brandname} includes so many configurable combinations that determining a single formula for performance metrics that covers all use cases is not possible.  
+
+Additionally, consider the following inter-related factors that apply to {brandname}:
+
+* Available CPU and memory resources in cloud environments
+* Caches used in parallel
+* Get, put, query balancing 
+* Peak load and throughput limitations
+* Querying limitations with data set
+* Number of entries per cache
+* Size of cache entries
+
+Given the number of different combinations and unknown external factors, providing a performance calculation that meets all {brandname} use cases is not possible, because you cannot compare one performance test to another test if any of the previously listed factors are different. 
+
+The purpose of this document is to provide details about use cases and architectures that can help you determine requirements for your {brandname} deployment.
+
+You can run a basic performance test from CLI that collects limited performance metrics. You can customize the performance test by parameters, so that the test outputs results that might meet your needs. The results from the test can provide you with baseline metrics that you can then use to determine settings and resources for your {brandname} caching requirements. 
```

Comment:
"from the {brandname} command line interface (CLI)"

Reply:
Also please take a look at the one sentence per line recommendation. This is something the Infinispan doc set adheres to. 

https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line

Reply:
Hi @oraNod . This recommendation is based on a document in draft, and the style does not seem to be evident on other projects I've worked on. I do find sticking to this recommendation difficult, because I don't know the "why" of it? The principle has not been explained to me very well. Is it recommended because it makes scanning an _.adoc_ file easier?

Reply:
The fact that doc is in some kind of draft state doesn't really matter. The advantages that doc lists hold true and explain the why.

In addition to all that, one sentence per line helps with PR reviews. But, generally, it also makes a lot of sense because of [how asciidoc creates paragraphs](https://docs.asciidoctor.org/asciidoc/latest/blocks/paragraphs/).

You can find this in the Infinispan contributor's guide here: https://infinispan.org/docs/stable/titles/contributing/contributing.html#documentation_guidelines_general

Reply:
Thanks for sharing, @oraNod . I really need to know "why", as I've been asked to implement things and being told they don't break anything, so why should they hold up the progress of a PR. 

I generally follow internal guidelines, so I can stay up-to-date with the "one voice". Who updates the Infinispan Contributor's guide? For example, "contrctions" have been relaxed. I generally don't use them, but they are no longer slapped with the non-recommended guidelines.

Reply:
@dfitzmau little things like this can have larger, cumulative effects down the road. Part of why I brought it up is that I see using ventilated prose (one sentence per line) as a good writing habit. It's not just a style guide thing.

I picked it up from one of the core devs on this project and try to evangelize it a little. I'd suggest going back and reading Dan Allen's take on the advantages. He pretty much says it all.

I'm not sure what you mean by "one voice"?

"Who updates the Infinispan Contributor's guide?" - The Infinispan team update the Contributor's Guide.
"For example, "contrctions" have been relaxed. I generally don't use them, but they are no longer slapped with the non-recommended guidelines." - Again, not sure what you mean by this.

Reply:
I think the topic of "preference" is highly debatable. Chat channels are devoted to what someone thinks is right or wrong. I prefer consistency, so that one can remain agile among products.

"simplifying" processes was also mentioned at a very high-level. Having too many sources to reference will eventually lead to confusion, as they start to contradict one another. 

The contributor guide you linked mentions "Avoid contractions. Use "do not" instead of "don’t".",